specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/cosmos-db.json

```diff
@@ -8861,6 +9137,322 @@
           "description": "The latest restorable timestamp for a resource."
         }
       }
+    },
+    "ServiceResource": {
+      "description": "Service cluster response resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The service name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        },
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a Data Transfer cluster.",
+          "$ref": "#/definitions/ServiceClusterProperty"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "ServiceClusterProperty": {
+      "description": "Describes the service cluster response property.",
```

Comment:
This mostly looks good, an interesting detail is the fact that for sqlDedicatedGateway ServiceResouce contains additional property called SqlDedicatedGateway enpoint. 

Here discriminator will come into picture. We will have two different resource models and depending on the value of ServiceName the resource model would be picked.

Reply:
We should also model ServiceName values as Enum.